### PR TITLE
test: check to have DataProviders code agnostic of PHP version

### PR DESCRIPTION
**Pull request by keradus**
depends on PRs raised few mins ago

### CODE REVIEW

**Comment by kubawerlos** [comment]
Isn't it an overkill? I was thinking of simply checking for:
```
                    [T_STRING, 'PHP_VERSION_ID'],
                    [T_STRING, 'PHP_MAJOR_VERSION'],
                    [T_STRING, 'PHP_MINOR_VERSION'],
                    [T_STRING, 'PHP_RELEASE_VERSION'],
                    [T_STRING, 'phpversion'],
```
in any test file, as all those should be handled via `@requires`, right?

**Review by kubawerlos — CHANGES_REQUESTED** [review]
Can we have it similar (https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7580) to how we check `preg_` functions calls?

**Comment by keradus** [comment]
We do not want `preg_` at all (except of one file - `Preg.php`), because we have our `Preg` class.

For `PHP_` in tests, I do not want to have it in data providers, but I'm OK to have it anywhere else. I think your approach is forbidding to use it anywhere else.
I would even consider to extend my approach to look not only within `provide*` class, but also within `if` conditions. Then, things like `tests/FixerDefinition/VersionSpecificationTest.php` won't need to be an exception

**Comment by kubawerlos** [comment]
> Then, things like `tests/FixerDefinition/VersionSpecificationTest.php` won't need to be an exception

Aren't those 2 functions the only places we want to allow PHP version awareness? IMHO in all other places `defined` should be enough.

**Comment by keradus** [comment]
I do not understand the part of `defined` in context of discussion where to use `PHP_*` constants.
I see no reason to prevent usage of them outside of data providers.

File: tests/AutoReview/ProjectCodeTest.php [hunk 1]
```diff
@@ -464,6 +465,91 @@ public function testExpectedInputOrder(string $testClassName): void
...
+                return false;
+            }
+
+            $nextToken = $tokens[$tokens->getNextMeaningfulToken($k)];
+
+            if (!$nextToken->isGivenKind(T_STRING) || !str_starts_with($nextToken->getContent(), 'provide')) {
+                // not a data provider method
+                return false;
+            }
+            // var_dump($testClassName, $exceptionMethods); die;
```
Comment (keradus) [inline tests/AutoReview/ProjectCodeTest.php:510]:
```suggestion

```

**Comment by kubawerlos** [comment]
> I do not understand the part of `defined` in context of discussion about where to use `PHP_*` constants.

Any `if (PHP_VERSION_*)` could/should be replaced with `if (defined('T_*')`.

> I see no reason to prevent usage of `PHP_*` outside of data providers (or more explicitly - conditions inside them).

I look at it form the other side - I see not reason to allow using `PHP_*` in tests.

**Comment by keradus** [comment]
Then, let's extract all `if(PHP_` and `if(defined` cleanups to #7581 , have it merged and see how many cases we have remaining, if any.

**Pull request merged**